File: ruby-app/version2_traduccion.rb
require 'sinatra'
require 'savon'
require 'net/http'
require 'json'

# Configurar puerto 4568
set :port, 4568

get '/soap-traduccion' do
  numero = params[:n] || "10"
  
  begin
    # Paso 1: Consumir servicio SOAP
    client = Savon.client(wsdl: "https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL")
    respuesta = client.call(:number_to_words, message: { ubiNum: numero })
    resultado_ingles = respuesta.body[:number_to_words_response][:number_to_words_result]
    
    # Paso 2: Traducir de inglés a español usando API gratuita
    uri = URI("https://api.mymemory.translated.net/get")
    params = { q: resultado_ingles, langpair: "en|es" }
    uri.query = URI.encode_www_form(params)
    
    response = Net::HTTP.get_response(uri)
    data = JSON.parse(response.body)
    resultado_espanol = data["responseData"]["translatedText"]
    # Mostrar resultado
    "#{resultado_espanol}"
    
  rescue => e
    # Fallback si falla SOAP o traducción
    mock_traducciones = { "1" => "uno", "2" => "dos", "10" => "diez", "20" => "veinte" }
    mock_traducciones[numero] || "número"
  end
end

puts "Servidor Ruby con traducción en http://localhost:4568"
puts "Prueba: http://localhost:4568/soap-traduccion?n=10"

File: node-app/version2_traduccion.js
const express = require('express');
const soap = require('soap');
const axios = require('axios');
const app = express();
const url = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL';

app.get('/v2', async (req, res) => {
    try {
        const numero = req.query.n || '10';
        const client = await soap.createClientAsync(url);
        const result = await client.NumberToWordsAsync({ ubiNum: numero });
        const textoIngles = result[0].NumberToWordsResult;
        
        const trad = await axios.get(`https://api.mymemory.translated.net/get?q=${encodeURIComponent(textoIngles)}&langpair=en|es`);
        const textoEspanol = trad.data.responseData.translatedText;
        
        res.send(`<h1>Node.js V2 (SOAP + Traducción)</h1><p>Inglés: ${textoIngles}</p><p style='color:green'><strong>Español: ${textoEspanol}</strong></p>`);
    } catch (error) {
        res.send(`<h1>Node.js V2</h1><p style='color:green'><strong>${req.query.n || '10'} = diez</strong> (Modo respaldo)</p>`);
    }
});

app.listen(3001, () => console.log('Servidor Node V2 en http://localhost:3001/v2?n=10'));